iothub/device/samples/convention-based-samples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -0,0 +1,391 @@
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    public class TemperatureControllerSample
+    {
+        private const string Thermostat1 = "thermostat1";
+        private const string Thermostat2 = "thermostat2";
+
+        private static readonly Random s_random = new Random();
+
+        private readonly DeviceClient _deviceClient;
+        private readonly ILogger _logger;
+
+        // Dictionary to hold the temperature updates sent over each "Thermostat" component.
+        // NOTE: Memory constrained devices should leverage storage capabilities of an external service to store this
+        // information and perform computation.
+        // See https://docs.microsoft.com/en-us/azure/event-grid/compare-messaging-services for more details.
+        private readonly Dictionary<string, Dictionary<DateTimeOffset, double>> _temperatureReadingsDateTimeOffset =
+            new Dictionary<string, Dictionary<DateTimeOffset, double>>();
+
+        // Dictionary to hold the current temperature for each "Thermostat" component.
+        private readonly Dictionary<string, double> _temperature = new Dictionary<string, double>();
+
+        // Dictionary to hold the max temperature since last reboot, for each "Thermostat" component.
+        private readonly Dictionary<string, double> _maxTemp = new Dictionary<string, double>();
+
+        public TemperatureControllerSample(DeviceClient deviceClient, ILogger logger)
+        {
+            _deviceClient = deviceClient ?? throw new ArgumentNullException($"{nameof(deviceClient)} cannot be null.");
+            _logger = logger ?? LoggerFactory.Create(builer => builer.AddConsole()).CreateLogger<TemperatureControllerSample>();
+        }
+
+        public async Task PerformOperationsAsync(CancellationToken cancellationToken)
+        {
+            // This sample follows the following workflow:
+            // -> Set handler to receive and respond to writable property update requests.
+            // -> Set handler to receive and respond to commands.
+            // -> Update device information on "deviceInformation" component.
+            // -> Send initial device info - "workingSet" over telemetry, "serialNumber" over reported property update - root interface.
+            // -> Periodically send "temperature" over telemetry - on "Thermostat" components.
+            // -> Send "maxTempSinceLastReboot" over property update, when a new max temperature is set - on "Thermostat" components.
+
+            _logger.LogDebug("Subscribe to writable property updates.");
+            await _deviceClient.SubscribeToWritablePropertiesEventAsync(HandlePropertyUpdatesAsync, null, cancellationToken);
+
+            _logger.LogDebug($"Subscribe to commands.");
+            await _deviceClient.SubscribeToCommandsAsync(HandleCommandsAsync, null, cancellationToken);
+
+            await UpdateDeviceInformationAsync(cancellationToken);
+            await SendDeviceSerialNumberAsync(cancellationToken);
```

Comment:
Maybe call this SendTelemetry as what we are showing in this is how to send telemetry?